Automattic/newspack-newsletters
# Add error and success notices

### All Submissions:






### Changes proposed in this Pull Request:



Adds error and success reporting for API calls.

Closes #24 
Closes #27 

### How to test the changes in this Pull Request:

1. Observe success notice after sending a test email
2. Send a test email to `a@example.com`. Observe an error notice displayed.
3. Exceed [test sending limits](https://mailchimp.com/help/about-test-email-sending-limits/) – observe an error notice.
4. Create a newsletter, save it. Then head over to Mailchimp and delete the corresponding campaign. Edit the newsletter and save draft – observe errors. 
5. Refresh page. After data is fetched, the error about synchronization problem should be displayed again.
6. Repeat steps 4, 5, but now try to publish instead of save. 
7. Create a new newsletter, add some content. Head over to settings and remove the Mailchimp API key. 
8. Try to save or publish the newsletter – observe error notices about missing API key. Add the key back in, click "Resync" button on the error notice – it should disappear and the settings should be filled in the sidebar.
9. Create a newsletter, fill all necessary information so it's valid. Schedule the post to publish in a few minutes. Then remove the corresponding campaign in Mailchimp. Make some tea, chill out, come back after a few minutes – the post should have been published by now. Visit the post edit screen – observe that an error is shown after data is fetched.

Because the campaign is always synchronized before sending, it's challenging to error out on actual _sending_. What you can do though, is edit the URL used in `send_campaign` function, so that it's invalid. Then try to publish a newsletter post and observe the error being reported.

### Other information:
